src/frontend/src/components/SearchModal.tsx

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useState } from "react";
+import { Dialog, Heading } from "react-aria-components";
+import { useTranslation } from "react-i18next";
+
+import {
+    selectIsSearchModalOpen,
+    setIsSearchModalOpen,
+} from "reducers/modalsDisplay/modalsDisplaySlice";
+import {
+    selectNeighborhoodFilters,
+    selectNeighborhoodFilterSuggestions,
+    setNeighborhoodFilters,
+} from "reducers/neighborhoods/neighborhoodsSlice";
+import { getRankedNeighborhoodLists } from "reducers/neighborhoods/neighborhoodsThunk";
+import type { FiltersState } from "reducers/neighborhoods/types";
+import { useAppDispatch, useAppSelector } from "store/store";
+
+import { Modal, ModalOverlay } from "components/base/Modal/Modal";
+import ModalCloseButton from "./ModalCloseButton";
+import {
+    Autocomplete,
+    type Suggestion,
+} from "./base/Autocomplete/Autocomplete";
+
+const SearchModal = ({ isMobile = false }) => {
```

Comment:
At the first glance, this component looks similar to `SearchList` but not actually the same after a closer look. I think there might be opportunity to further componentize one of them and reuse. Due to time and budget, I think we should defer it though.